ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -194,8 +199,9 @@ internal constructor(
     // into the repo
     // LOIs that are persisted to the local and remote dbs.
     val loiStream =
-      surveyRepository.activeSurvey.switchMap { activeProject ->
-        getLocationsOfInterestStream(activeProject)
+      surveyRepository.activeSurvey.switchMap { survey ->
+        activeSurveyId = survey.map { it.id }.orElse("")
+        getLocationsOfInterestStream(survey)
```

Comment:
We already have a class which provides this stream (LoiCardSource). Maybe we can rename that to something more generic and reuse it here?

Reply:
This is going away in an upcoming PR; LOIs will be synced in a background worker instead. Added a TODO to clarify.